pkg/rehearse/rehearse.go

```diff
@@ -630,13 +636,91 @@ func ensureImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client,
 				return fmt.Errorf("failed waiting for imagestreamtag %s to appear: %w", requiredImageStreamTag, err)
 			}
 
+			if err := ensureImageInQCI(ctx, client, requiredImageStreamTag, registryConfig, istLog); err != nil {
+				log.WithError(err).Errorf("failed to mirror imagestreamtag %s", requiredImageStreamTag)
+			}
+
 			return nil
 		})
 	}
 
 	return g.Wait()
 }
 
+func ensureImageInQCI(ctx context.Context, client ctrlruntimeclient.Client, requiredImageStreamTag types.NamespacedName, registryConfig string, log *logrus.Entry) error {
```

Comment:
I would also try to dry out a function shared by this tool and `ci-images-mirror`. They look quite similar to each other.